lib/packwerk.rb

```diff
@@ -14,8 +14,11 @@ module Packwerk
   extend ActiveSupport::Autoload
 
   # Public APIs
+  autoload :ApplicationLoadPaths
```

Comment:
I stub `ApplicationLoadPaths` so that I can run `packwerk` in some integration tests for some things in the broader packwerk ecosystem.

Reply:
Can't you just stub the caller method in configuration? Why specifically this module?

Reply:
I probably can -- I'll look into this!

Reply:
So... I use `ApplicationLoadPaths` in `danger-packwerk` to initialize a `ConstantResolver`: https://github.com/rubyatscale/danger-packwerk/blob/cbcd07cf42f7f555e02e0b1ac36552142a4262d4/lib/danger-packwerk/private.rb#L17, which we use to find where a constant is defined: https://github.com/rubyatscale/danger-packwerk/blob/cbcd07cf42f7f555e02e0b1ac36552142a4262d4/lib/danger-packwerk/danger_package_todo_yml_changes.rb#L121 for the purpose of being able to link to constant definitions when we leave GitHub comments through our Danger integration.

I understand it's not exactly the intended public API of packwerk, and I always could just copy and paste this code into `danger-packwerk` for it to get the load paths itself, but wondering if this would be alright to use until we find a better solution. Also open to other suggestions you have!

Reply:
I have an issue making `ApplicationLoadPaths` public for 3 reasons:

- I think it is better to name it `LoadPaths.for(root, evironment:)` or similar.
- It looks like we are exposing public methods we don't need to in it. `extract_relevant_paths` should be the only public method.
- We use a railties as a peer dependency (we reference `Rails` but don't depend on railties or require it anywhere). This access should be safer and tolerant of it not being there. Or, we should use proper peer dependency syntax (see [example](https://github.com/Shopify/rails/blob/656118bc17791291db7c10075b489dfc6f93fce3/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L3-L4)).

Reply:
These are valid concerns. How do you feel about us merging this PR as is and I do a fast follow with these updates? Happy to do first if you'd prefer.

Reply:
Approved, we can do this next.